lib/rqrcode_core/qrcode/qr_code.rb

```diff
@@ -485,6 +496,27 @@ def smallest_size_for(string, max_size_array) #:nodoc:
       ver + 1
     end
 
+    def smallest_size_for_multi(data:, level:, max_version: 40, min_version: 1)
+      raise QRCodeArgumentError, "Data too long for QR Code" if min_version > max_version
+
+      # Manually calculate max size
+      # rs_blocks = QRRSBlock.get_rs_blocks(min_version, QRERRORCORRECTLEVEL[level])
+      # max_size_bits = QRCode.count_max_data_bits(rs_blocks)
+
```

Comment:
I used the same table you referenced in the comments to create the `QRMAXBITS` list. Because multi-mode support can have any number of encodings, I count the total number of bits between all the segments and compare it to the maximum number of bits supported by the level/version. If there is a way to reuse `QRMAXDIGITS` for an arbitrary number of encoded segments, then I can remove it, but I couldn't figure that out. Theoretically you could use `QRMAXBITS` for everything, but I didn't want to touch the existing code.

Reply:
Thanks for the explanation. Any change that will also improve the existing code base would be welcome. The test suite in this lib and RQRCode is quite comprehensive so any problems should be easy to spot.

Reply:
OK, I can remove `QRMAXDIGITS` and just use the `QRMAXBITS` calculation all around if you want.

Reply:
Thanks. Maybe the complete removal should be left to another follow up PR? 

I'm away with my family for a week but I'll look at all this as soon as I am back.

Reply:
Enjoy your time off, this should be ready when you get back.